pkg/datasource/consul/consul.go

```diff
@@ -107,9 +108,9 @@ func (c *consulDataSource) watch() {
 				time.Sleep(time.Second)
 				continue
 			}
-			logging.FrequentErrorOnce.Do(func() {
+			if ok := atomic.CompareAndSwapUint32(&logging.FrequentErrorOnce, 0, 1); ok {
```

Comment:
logging.FrequentErrorOnce used in multiple places, one place cas changes to 1, the others are never executed

Reply:
yes，same as  `logging.FrequentErrorOnce.Do` .  once ` logging.FrequentErrorOnce.Do` was executed , the others are never executed